src/format_changeset.rs

```diff
@@ -0,0 +1,70 @@
+use difference::{Difference, Changeset};
+use std::fmt;
+use ansi_term::Colour::{Red, Green};
+use ansi_term::Style;
+
+macro_rules! paint {
+    ($f: ident, $colour: ident, $fmt: expr, $($args:tt)*) => (
+        write!($f, "{}", $colour.paint(format!($fmt, $($args)*)))
+    )
+}
+
+
+// Adapted from:
+// https://github.com/johannhof/difference.rs/blob/c5749ad7d82aa3d480c15cb61af9f6baa08f116f/examples/github-style.rs
+// Credits johannhof (MIT License)
```

Comment:
@fflorent I think we cannot adopt code from the difference crate, since we use dual-license MIT+Apache and the [difference crate does not use the Apache license (#14)](https://github.com/johannhof/difference.rs/issues/14).

I see these alternatives:
- ask @johannhof for permission to use this code in our crate with MIT+Apache, even if the difference crate still uses MIT only
- rewrite this part from scratch (it's not mutch code)
- extend the display functionality of the difference crate via a separate PR